### PR TITLE
networkfirewall: Add 4 resources, networkmanager: Add 9 resources

### LINKED ISSUE #59: Configure networkfirewall (4),networkmanager (9)
External name configuration of 4 resources in the `networkfirewall` group:
* [`aws_networkfirewall_firewall_policy`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkfirewall_firewall_policy)
* [`aws_networkfirewall_logging_configuration`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkfirewall_logging_configuration)
* [`aws_networkfirewall_resource_policy`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkfirewall_resource_policy)
* [`aws_networkfirewall_rule_group`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkfirewall_rule_group)

External name configuration of 9 resources in the `networkmanager` group:
* [`aws_networkmanager_connection`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_connection)
* [`aws_networkmanager_customer_gateway_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_customer_gateway_association)
* [`aws_networkmanager_device`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_device)
* [`aws_networkmanager_global_network`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_global_network)
* [`aws_networkmanager_link`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_link)
* [`aws_networkmanager_link_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_link_association)
* [`aws_networkmanager_site`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_site)
* [`aws_networkmanager_transit_gateway_connect_peer_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_transit_gateway_connect_peer_association)
* [`aws_networkmanager_transit_gateway_registration`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_transit_gateway_registration)

### PR DESCRIPTION
**Pull request by dverveiko**
### Description of your changes
Added configuration for 4 resources in `networkfirewall` group:
- [x] [`aws_networkfirewall_firewall_policy`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkfirewall_firewall_policy)
- [x] [`aws_networkfirewall_logging_configuration`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkfirewall_logging_configuration)
- [x] [`aws_networkfirewall_resource_policy`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkfirewall_resource_policy)
- [x] [`aws_networkfirewall_rule_group`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkfirewall_rule_group)

Added configuration for 9 resources in `networkmanager` group:
 [`aws_networkmanager_connection`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_connection)
- [x] [`aws_networkmanager_customer_gateway_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_customer_gateway_association)
- [x] [`aws_networkmanager_device`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_device)
- [x] [`aws_networkmanager_global_network`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_global_network)
- [x] [`aws_networkmanager_link`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_link)
- [x] [`aws_networkmanager_link_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_link_association)
- [x] [`aws_networkmanager_site`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_site)
- [x] [`aws_networkmanager_transit_gateway_connect_peer_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_transit_gateway_connect_peer_association)
- [x] [`aws_networkmanager_transit_gateway_registration`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_transit_gateway_registration)

Fixes #59 

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### CODE REVIEW

File: config/externalnamenottested.go [hunk 1]
```diff
@@ -751,4 +751,48 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
 	// aws_storagegateway_working_storage can be imported by using the gateway Amazon Resource Name (ARN) and local disk identifier separated with a colon (:)
 	// Example: arn:aws:storagegateway:us-east-1:123456789012:gateway/sgw-12345678:pci-0000:03:00.0-scsi-0:0:0:0
 	"aws_storagegateway_working_storage": config.TemplatedStringAsIdentifier("", "{{ .parameters.gateway_arn }}:{{ .parameters.disk_id }}"),
+
+	// networkfirewall
+	//
+	// Network Firewall Policies can be imported using their ARN
+	// Example: arn:aws:network-firewall:us-west-1:123456789012:firewall-policy/example
+	"aws_networkfirewall_firewall_policy": config.TemplatedStringAsIdentifier("", "arn:aws:network-firewall:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:firewall-policy/{{ .external_name }}"),
```
Comment (sergenyalcin) [inline config/externalnamenottested.go:759]: We need to use `name` as the `externalNamePath` parameter of the function. 

```suggestion
	"aws_networkfirewall_firewall_policy": config.TemplatedStringAsIdentifier("name", "arn:aws:network-firewall:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:firewall-policy/{{ .external_name }}"),
```

File: config/externalnamenottested.go [hunk 2]
```diff
@@ -751,4 +751,48 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+	"aws_networkfirewall_resource_policy": config.TemplatedStringAsIdentifier("", "arn:aws:network-firewall:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:stateful-rulegroup/{{ .external_name }}"),
+	// Network Firewall Rule Groups can be imported using their ARN
+	// Example: arn:aws:network-firewall:us-west-1:123456789012:stateful-rulegroup/example
+	"aws_networkfirewall_rule_group": config.TemplatedStringAsIdentifier("", "arn:aws:network-firewall:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:stateful-rulegroup/{{ .external_name }}"),
+
+	// networkmanager
+	//
+	// aws_networkmanager_connection can be imported using the connection ARN
+	// Example: arn:aws:networkmanager::123456789012:device/global-network-0d47f6t230mz46dy4/connection-07f6fd08867abc123
+	"aws_networkmanager_connection": config.TemplatedStringAsIdentifier("", "arn:aws:networkmanager::{{ .setup.client_metadata.account_id }}:device/{{ .parameters.global_network_id }}/{{ .external_name }}"),
```
Comment (sergenyalcin) [inline config/externalnamenottested.go:774]: I am not sure that, the last element of the template string will provide this string: `connection-07f6fd08867abc123`

I will suggest two things for this resource:

- We may consider using the existing configuration and let's see what is happening while testing.
- Or we may consider using IdentifierFromProvider here.

Decision is yours @dverveiko

File: config/externalnamenottested.go [hunk 3]
```diff
@@ -751,4 +751,48 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+	// aws_networkmanager_customer_gateway_association can be imported using the global network ID and customer gateway ARN
+	// Example: global-network-0d47f6t230mz46dy4,arn:aws:ec2:us-west-2:123456789012:customer-gateway/cgw-123abc05e04123abc
+	"aws_networkmanager_customer_gateway_association": config.TemplatedStringAsIdentifier("", "{{ .parameters.global_network_id }},arn:aws:ec2:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:customer-gateway/{{ .external_name }}"),
```
Comment (sergenyalcin) [inline config/externalnamenottested.go:777]:
```suggestion
	"aws_networkmanager_customer_gateway_association": config.TemplatedStringAsIdentifier("", "{{ .parameters.global_network_id }},arn:aws:ec2:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:customer-gateway/{{ .parameters.customer_gateway_arn}}"),
```

File: config/externalnamenottested.go [hunk 4]
```diff
@@ -751,4 +751,48 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+	// aws_networkmanager_device can be imported using the device ARN
+	// Example: arn:aws:networkmanager::123456789012:device/global-network-0d47f6t230mz46dy4/device-07f6fd08867abc123
+	"aws_networkmanager_device": config.TemplatedStringAsIdentifier("", "arn:aws:networkmanager::{{ .setup.client_metadata.account_id }}:device/{{ .parameters.global_network_id }}/{{ .external_name }}"),
```
Comment (sergenyalcin) [inline config/externalnamenottested.go:780]: Same concern in the `aws_networkmanager_connection`

File: config/externalnamenottested.go [hunk 5]
```diff
@@ -751,4 +751,48 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+	// aws_networkmanager_global_network can be imported using the global network ID
+	"aws_networkmanager_global_network": config.IdentifierFromProvider,
+	// aws_networkmanager_link can be imported using the link ARN
+	// Example: arn:aws:networkmanager::123456789012:link/global-network-0d47f6t230mz46dy4/link-444555aaabbb11223
+	"aws_networkmanager_link": config.TemplatedStringAsIdentifier("", "arn:aws:networkmanager::{{ .setup.client_metadata.account_id }}:link/{{ .parameters.global_network_id }}/{{ .external_name }}"),
```
Comment (sergenyalcin) [inline config/externalnamenottested.go:785]: Same concern in the aws_networkmanager_connection

File: config/externalnamenottested.go [hunk 6]
```diff
@@ -751,4 +751,48 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+	// aws_networkmanager_link_association can be imported using the global network ID, link ID and device ID
+	// Example: global-network-0d47f6t230mz46dy4,link-444555aaabbb11223,device-07f6fd08867abc123
+	"aws_networkmanager_link_association": config.TemplatedStringAsIdentifier("", "{{ .parameters.global_network_id }},{{ .parameters.link_id }},{{ .parameters.device_id }}"),
+	// aws_networkmanager_site can be imported using the site ARN
+	// Example: arn:aws:networkmanager::123456789012:site/global-network-0d47f6t230mz46dy4/site-444555aaabbb11223
+	"aws_networkmanager_site": config.TemplatedStringAsIdentifier("", "arn:aws:networkmanager::{{ .setup.client_metadata.account_id }}:site/{{ .parameters.global_network_id }}/{{ .external_name }}"),
```
Comment (sergenyalcin) [inline config/externalnamenottested.go:791]: Same concern in the aws_networkmanager_connection

File: config/externalnamenottested.go [hunk 7]
```diff
@@ -751,4 +751,48 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+	// aws_networkmanager_transit_gateway_connect_peer_association can be imported using the global network ID and customer gateway ARN
+	// Example: global-network-0d47f6t230mz46dy4,arn:aws:ec2:us-west-2:123456789012:transit-gateway-connect-peer/tgw-connect-peer-12345678
+	"aws_networkmanager_transit_gateway_connect_peer_association": config.TemplatedStringAsIdentifier("", "{{ .parameters.global_network_id }},arn:aws:ec2:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:transit-gateway-connect-peer/{{ .external_name }}"),
```
Comment (sergenyalcin) [inline config/externalnamenottested.go:794]:
```suggestion
	"aws_networkmanager_transit_gateway_connect_peer_association": config.TemplatedStringAsIdentifier("", "{{ .parameters.global_network_id }},arn:aws:ec2:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:transit-gateway-connect-peer/{{ .parameters.transit_gateway_connect_peer_arn }}"),
```

**Review by sergenyalcin — COMMENTED** [review]
Thanks for this PR @dverveiko I left a few comments

File: config/externalnamenottested.go (hunk 2 above)
Comment (dverveiko) [reply config/externalnamenottested.go:755]: To better understanding of configuration process, I've decided to leave this configuration as is and check what will happen in testing. I'll leave a comment for that person who will test this.

**Review by sergenyalcin — APPROVED** [review]
Thanks @dverveiko LGTM!

**Pull request merged**